src/Core/ViewHelper/AbstractTagBasedViewHelper.php

```diff
@@ -88,6 +88,9 @@ public function initializeArguments()
     public function initialize()
     {
         parent::initialize();
+        $this->tag->reset();
+        $this->tag->setTagName($this->tagName);
```

Comment:
I wonder if this could be moved to `setTagBuilder()` and drop the `$this->tagName` injection from the constructor ...

Reply:
Rather, I'd remove the line that sets the tag name in `setTagBuilder` since setter methods shouldn't mutate the object that is passed.

TagBuilders are already initialized with tag name as constructor argument ;)

In the future, tag based VH should become compilable in a way that TagBuilder is constructed outside and has one instance per execution, then passed to a static method that will render the tag. Then the TagBuilder reset and mutability can be reworked. Point being, if we aim for that then anything that treats the TagBuilder instance as a more stupid thing that needs init on every call, is imho a good thing. Even if it didn't also solve a bug ;)